src/scripts/bench.py

```diff
@@ -20,6 +17,12 @@
 import subprocess
 import re
 import json
+import math
+import platform
+from datetime import datetime
+import shutil
+import numpy as np
+import matplotlib.pyplot as plt
```

Comment:
Maybe import those optionally, so that numpy and matplotlib isn't a hard dependency for this script?

Along those lines:

```suggestion
try:
    import numpy as np
except:
    pass
```

... w/o those dependencies we could just not provide the `--html-report` option. No strong opinion either way, from my side, though.

Reply:
No strong opinion from me either. For scripts the end user has to run to build/test the library, stock Python only (or with purely optional use of modules).

For scripts that are informational or for developer use, as here, anything goes.

Reply:
Thanks for the input. I'll leave it as-is.